lib/swift/Sources/TSocketServer.swift

```diff
@@ -96,34 +96,29 @@ open class TSocketServer<InProtocol: TProtocol, OutProtocol: TProtocol, Processo
 
     // throw away our socket
     CFSocketInvalidate(sock)
-
-    // register for notifications of accepted incoming connections
-    _ = NotificationCenter.default.addObserver(forName: .NSFileHandleConnectionAccepted,
-                                               object: nil, queue: nil) {
-                                                [weak self] notification in
-                                                guard let strongSelf = self else { return }
-                                                guard let clientSocket = notification.userInfo?[NSFileHandleNotificationFileHandleItem] as? FileHandle else { return }
-                                                strongSelf.connectionAccepted(clientSocket)
-    }
-
-    // tell socket to listen
-    socketFileHandle.acceptConnectionInBackgroundAndNotify()
-
+    
     print("TSocketServer: Listening on TCP port \(port)")
+    
+    // tell socket to listen
+    acceptConnectionInBackgroundAndNotify(handle: socketFileHandle)
   }
-
-  deinit {
-    NotificationCenter.default.removeObserver(self)
+  
+  open func acceptConnectionInBackgroundAndNotify(handle: FileHandle) {
+    DispatchQueue(label: "ConnectionAcceptQueue").async {
+      let acceptedFD = accept(handle.fileDescriptor, nil, nil)
+      DispatchQueue.main.async {
```

Comment:
maybe instead of hard coding DispatchQueue this could be a public property? maybe a client would want to use a custom queue?

Reply:
I don't think this should be exposed. We have to do it this way rn because NotificationCenter doesn't seem to be delivering notifications on Linux targets. We'd probably prefer to use the system implementation of `FileHandle. acceptConnectionInBackgroundAndNotify` and so in the future if this is fixed on Linux in the CoreFoundation implementation then we might want to move back to that and having this queue exposed as public API would be a breaking change.

Reply:
The `byte` type was **always** signed, like all Thrift integer types. So we renamed the type specifier a while ago from byte to `i8` mostly to remove that constant source of confusion - a byte is considered unsigned in many languages. Nevertheless, for compatibility reasons you still can use `byte` in the IDL.